### PR TITLE
Maskable eval callback call callback fix

### LINKED ISSUE #92: MaskableEvalCallback doesn't work with StopTrainingOnNoModelImprovement
**Describe the bug**
MaskableEvalCallback doesn't call StopTrainingOnNoModelImprovement. It's because it's only called on new best mean reward, instead of after every evaluation. In those cases StopTrainingOnNoModelImprovement always resets it's no_improvement_evals, so it never stops the training.
Currently, this is the end of the step of MaskableEvalCallback:
[MaskableEvalCallback ](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/3cbd2429be7cf94d40c72e1e8801743345999ace/sb3_contrib/common/maskable/callbacks.py#L97)
```python
            if mean_reward > self.best_mean_reward:
                if self.verbose > 0:
                    print("New best mean reward!")
                if self.best_model_save_path is not None:
                    self.model.save(os.path.join(self.best_model_save_path, "best_model"))
                self.best_mean_reward = mean_reward
                # Trigger callback if needed
                if self.callback is not None:
                    return self._on_event()
```
Instead it should call self.callback_on_new_best on new best mean reward instead and self.callback after every evaluation:
```python
            continue_training = True

            if mean_reward > self.best_mean_reward:
                if self.verbose > 0:
                    print("New best mean reward!")
                if self.best_model_save_path is not None:
                    self.model.save(os.path.join(self.best_model_save_path, "best_model"))
                self.best_mean_reward = mean_reward
                # Trigger callback on new best model, if needed
                if self.callback_on_new_best is not None:
                    continue_training = self.callback_on_new_best.on_step()

            # Trigger callback after every evaluation, if needed
            if self.callback is not None:
                continue_training &= self._on_event()

            return continue_training 
```
Just like in [EvalCallback](https://github.com/DLR-RM/stable-baselines3/blob/fda3d4d748439a6755260896e4350e0383a9c6f1/stable_baselines3/common/callbacks.py#L444)

**Code example**
```python
from sb3_contrib import MaskablePPO
from sb3_contrib.common.maskable.callbacks import MaskableEvalCallback
from stable_baselines3.common.callbacks import StopTrainingOnNoModelImprovement
from minigames.collect_minerals_and_gas.src.env_dicrete import CollectMineralAndGasDiscreteEnv


env = CollectMineralAndGasDiscreteEnv(step_mul=8, realtime=False)
stop_callback = StopTrainingOnNoModelImprovement(max_no_improvement_evals=10, min_evals=10, verbose=1)
eval_callback = MaskableEvalCallback(
    env, eval_freq=100000, deterministic=False, render=False, callback_after_eval=stop_callback, n_eval_episodes=20
)

model = MaskablePPO("MlpPolicy", env)
model.learn(10000000, callback=eval_callback)
```

**System Info**
Describe the characteristic of your environment:
 * Describe how the library was installed (pip, docker, source, ...) **PIP**
 * Stable-Baselines3 and sb3-contrib versions **1.5.0, contrib: 1.5.0**
 * GPU models and configuration **RTX 3070**
 * Python version **3.9.7**
 * PyTorch version **1.11.0+cu113**
 * Gym version **0.21.0**

### PR DESCRIPTION
**Pull request by CppMaster**
closes #92 
## Checklist:
- [x] I've read the [CONTRIBUTION](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] The functionality/performance matches that of the source (**required** for new training algorithms or training-related features).
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have included an example of using the feature (*required for new features*).
- [ ] I have included baseline results (**required** for new training algorithms or training-related features).
- [x] I have updated the documentation accordingly. (docstrings)
- [x] I have updated the changelog accordingly (**required**).
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

I cannot run commands 'make *', because I work on Windows and it's not installed

### CODE REVIEW

**Comment by araffin** [comment]
Hello,
thanks for the PR, please fill up the template completely (notably the checklist)

**Comment by CppMaster** [comment]
Checklist included

File: sb3_contrib/ppo_mask/ppo_mask.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from stable_baselines3.common.buffers import RolloutBuffer
 from stable_baselines3.common.callbacks import BaseCallback, CallbackList, ConvertCallback
 from stable_baselines3.common.on_policy_algorithm import OnPolicyAlgorithm
-from stable_baselines3.common.policies import BasePolicy
+from stable_baselines3.common.policies import BasePolicy, register_policy
```
Comment (araffin) [inline sb3_contrib/ppo_mask/ppo_mask.py:14]: it seems you are using old version of SB3, `register_policy` was removed in 1.6.0

File: tests/test_invalid_actions.py [hunk 2]
```diff
@@ -191,6 +192,24 @@ def test_callback(tmp_path):
...
 
 
+def test_child_callback():
+    """
+    Stop callback and callback on new best rewards
+    """
+
+    env = make_env()
+    eval_env = make_env()
+    model = MaskablePPO("MlpPolicy", env)
```
Comment (araffin) [inline tests/test_invalid_actions.py:202]:
```suggestion
    model = MaskablePPO("MlpPolicy", env, n_steps=64)
```

File: tests/test_invalid_actions.py [hunk 3]
```diff
@@ -191,6 +192,24 @@ def test_callback(tmp_path):
...
+    stop_callback = StopTrainingOnNoModelImprovement(1, 2)
+    new_best_mean_callback = EventCallback()
+    eval_callback = MaskableEvalCallback(Monitor(eval_env), eval_freq=1, callback_after_eval=stop_callback,
```
Comment (araffin) [inline tests/test_invalid_actions.py:205]:
```suggestion
    eval_callback = MaskableEvalCallback(Monitor(eval_env), eval_freq=64, callback_after_eval=stop_callback,
```

File: tests/test_invalid_actions.py [hunk 4]
```diff
@@ -191,6 +192,24 @@ def test_callback(tmp_path):
...
+                                         callback_on_new_best=new_best_mean_callback)
+    model.learn(1000, callback=eval_callback)
```
Comment (araffin) [inline tests/test_invalid_actions.py:207]:
```suggestion
    model.learn(128, callback=eval_callback)
```

tests should run as fast as possible

File: docs/misc/changelog.rst [hunk 5]
```diff
@@ -17,6 +17,7 @@ Bug Fixes:
 ^^^^^^^^^^
 - Fixed the issue of wrongly passing policy arguments when using CnnLstmPolicy or MultiInputLstmPolicy with ``RecurrentPPO`` (@mlodel)
 - Fixed division by zero error when computing FPS when a small number of time has elapsed in operating systems with low-precision timers.
+- Fixed calling child callbacks in MaskableEvalCallback
```
Comment (araffin) [inline docs/misc/changelog.rst:20]:
```suggestion
- Fixed calling child callbacks in MaskableEvalCallback (@CppMaster)
```

**Comment by CppMaster** [comment]
@araffin Suggested changes implemented. I've also changed required version of stable-baseline3 to >=1.6.0,<1.7.0 so it doesn't break, when 1.7.0 is released (I don't expect minor releases to break anything)

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**